src/Elliptic/Protocols/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,207 @@
+ * systems we generally use the canonical system-specific symbols for the fields
+ * in place of these indices. See the `Poisson::FirstOrderSystem` and the
+ * `Elasticity::FirstOrderSystem` for examples.
+ *
+ * Conforming classes must have these static member variables:
+ *
+ * - `size_t volume_dim`: The number of spatial dimensions.
+ *
+ * Conforming classes must have these type aliases:
+ *
+ * - `primal_fields`: A list of tags representing the primal fields. These are
+ *   the fields we solve for, e.g. \f$u\f$ for a Poisson equation.
+ *
+ * - `auxiliary_fields`: A list of tags representing the auxiliary fields, which
+ *   are typically gradients of the primal fields, e.g. \f$v_i = \partial_i u\f$
+ *   for a Poisson equation. These must follow the order of the `primal_fields`.
+ *   Specifically, each auxiliary field must have one rank higher than its
+ *   corresponding primal field.
+ *
+ * - `primal_fluxes`: A list of tags representing the primal fluxes
+ *   \f$F_{u_A}^i\f$. These are typically some linear combination of the
+ *   auxiliary fields with raised indices, e.g. \f$v^i = g^{ij}v_j\f$ for a
+ *   curved-space Poisson equation. They must have the same rank as the
+ *   `auxiliary_fields` but with an upper-spatial first index, because their
+ *   divergence defines the elliptic equation.
+ *
+ * - `auxiliary_fluxes`: A list of tags representing the auxiliary fluxes
+ *   \f$F_{v_A}^i\f$, e.g. \f$\delta^i_j u\f$ for a Poisson equation. These must
+ *   have one rank higher than the `auxiliary_fields` and have an upper-spatial
+ *   first index because their divergence defines the auxiliary fields.
+ *
+ * - `background_fields`: A list of tags representing the variable-independent
+ *   background fields in the equations. Examples are a background metric,
+ *   associated fixed geometry quantities such as Christoffel symbols or the
+ *   Ricci scalar, or any other fixed field that determines the problem to be
+ *   solved such as matter sources in the Einstein constraint equations.
+ *
+ * - `inv_metric_tag`: The tag that defines the background geometry. This is the
+ *   metric responsible for normalizing one-forms, such as face normals.
+ *
+ * - `fluxes_computer`: A class that defines the primal and auxiliary fluxes
+ *   \f$F_\alpha^i\f$. Must have an `argument_tags` type alias and two `apply`
+ *   function overloads: One that computes the primal fluxes and another that
+ *   computes the auxiliary fluxes. The first `apply` function takes these
+ *   arguments in this order:
+ *
+ *   1. The `primal_fluxes` as not-null pointer
+ *   2. The `argument_tags`
+ *   3. The `auxiliary_fields`
+ *
+ *   The second `apply` function takes these arguments in this order:
+ *
+ *   1. The `auxiliary_fluxes` as not-null pointer
+ *   2. The `argument_tags`
+ *   3. The `primal_fields`
+ *
+ *   The functions can assume the output buffers are already correctly sized.
```

Comment:
is there any guarantee on the value at input? E.g. will they be zeroed?

Reply:
there is not, I added a comment 👍